downloader/download.go

```diff
@@ -237,14 +270,25 @@ func createRootedRepo(
 		return nil, err
 	}
 
-	if err := recursiveCopy(
-		"/", repo.FS(),
-		clonedPath, clonedFS,
-	); err != nil {
-		return nil, err
+	done := make(chan struct{})
+	go func() {
+		err = recursiveCopy(
+			"/", repo.FS(),
+			clonedPath, clonedFS,
+		)
+
+		done <- struct{}{}
```

Comment:
Same as with `lib.Hash` what happens if copy takes too much time. Is this going to block?

Reply:
yes, it will. Let's make the channel buffered with capacity 1